dbms/src/Storages/Page/V3/BlobStore.h

```diff
@@ -64,14 +64,23 @@ class BlobStore : private Allocator<false>
 
     FileUsageStatistics getFileUsageStatistics() const;
 
-    std::vector<BlobFileId> getGCStats();
-
-    PageEntriesEdit gc(GcEntriesMap & entries_need_gc,
-                       const PageSize & total_page_size,
+    using PageTypeAndBlobIds = std::map<PageType, std::vector<BlobFileId>>;
+    PageTypeAndBlobIds getGCStats();
+
+    void gc(
+        PageType page_type,
+        const GcEntriesMap & entries_need_gc,
+        const PageSize & total_page_size,
+        PageEntriesEdit & edit,
+        const WriteLimiterPtr & write_limiter = nullptr,
+        const ReadLimiterPtr & read_limiter = nullptr);
+
+    using PageTypeAndGcInfo = std::vector<std::tuple<PageType, GcEntriesMap, PageSize>>;
+    PageEntriesEdit gc(const PageTypeAndGcInfo & page_type_and_gc_info,
                        const WriteLimiterPtr & write_limiter = nullptr,
                        const ReadLimiterPtr & read_limiter = nullptr);
 
-    PageEntriesEdit write(typename Trait::WriteBatch && wb, const WriteLimiterPtr & write_limiter = nullptr);
+    PageEntriesEdit write(typename Trait::WriteBatch && wb, const WriteLimiterPtr & write_limiter = nullptr, PageType page_type = PageType::Normal);
```

Comment:
This should be better.

```suggestion
    PageEntriesEdit write(typename Trait::WriteBatch && wb, PageType page_type = PageType::Normal, const WriteLimiterPtr & write_limiter = nullptr);
```

Reply:
Fixed.